### ISSUE TITLE
lines-before-block also triggers for first constituent in a union or intersection type

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [gajus/eslint-plugin-jsdoc#1343]
## Expected behavior

This should not report:

```ts
export type SomeType =
	/**
	 * 🍏 fruit.
	 */
	| Apple

	/**
	 * 🍌 fruit.
	 */
	| Banana;
```

## Actual behavior

```plaintext
Required 1 line(s) before JSDoc blockeslint[jsdoc/lines-before-block](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/lines-before-block.md#repos-sticky-header)
```

## ESLint Config

```js
import jsdoc from 'eslint-plugin-jsdoc';

export default [jsdoc.configs['flat/recommended']];
```

## Additional info

I'd fixed this bug/feature for _blocks_ in #1341, but didn't think to for union or intersection types.

The autofix puts a blank line before the JSDoc block. Prettier removes that blank line. Fun.

**Comment by LukeAbby** [comment]
Fixed by #1383. Wasn't closed because I couldn't link it in my PR. @brettz9 could you close this?
